internal/schema/schema.go

```diff
@@ -170,6 +188,36 @@ func (s *Schema) LookupQueryTypesByFieldPath(fieldPath []string) ([]*Type, error
 	return types, nil
 }
 
+// GetInputFieldsForQueryPath is intended to return the fields that are
+// available as input arguments when performing a query using the received
+// query path.  For example, a []string{"actor", "account"} would look at
+// `actor { account { ...` for the representivate types at those schema levels
+// and determine collect the available arguments for return.
+func (s *Schema) GetInputFieldsForQueryPath(queryPath []string) map[string][]Field {
+	fields := make(map[string][]Field)
+
+	pathTypes, err := s.LookupQueryTypesByFieldPath(queryPath)
+	if err != nil {
+		log.Error(err)
```

Comment:
Should this return?

Reply:
Oh it could, but I think since its populated below and returned its probably fine.